### PR TITLE
v4.0.0

### PR DESCRIPTION
**Pull request by dhershman1**
## Breaking Changes

- Skip is no longer case sensitive `# skip` now works the same as `# SKIP` for tests
- Using asserts as the test counter now, this should prevent tap output using `#` for comments from recording tests that are actually comments
  - Addresses #23 partly (names are a bit wonky)
  - A deeper solution is needed for full compatibility possibly bringing the parser in house instead of relying on tap-out since it's focused on tape only output
- If a file extension is set in the command, it will be used for the generated file, if not `.xml` will be used
  - Addresses #24 
  - Example: `tap-junit -n tap.xuni` will now generate a tap.xuni file instead of a tap.xml file
  - Default is still `.xml` however

## New

- #24 Extension support (Listed in breaking changes)

## Notes

TL;DR: The problem in #23 is deeper than this solution, so some adjustments will need to be made because of the way some tools output their tap information.

As mentioned a bigger solution may be needed to address the problem within #23 itself. The parser needs to be more overall tap friendly rather than specific to how tape expects things to work. As this problem also persists within normal tap output.

For example, if I had this test:

```js
test('a + b = ab', t => {
  t.same('a' + 'b', 'ab')
  t.end()
})
```

The syntax for **tape** would be:

```
TAP version 13
# a + b = ab
ok 1 should be equivalent
1..1
# tests 1
# pass 1

# ok
```

while the syntax in normal Tap might be:

```
ok 1 - should be equivalent
1..1
```

For the most part, tap-out does a good job of picking up the differences here and we are able to parse out similar outputs. The problems lies in when something like [bats](https://github.com/sstephenson/bats) or anyone really, wants to add extra comments.

```
# I am a comment
ok 1 - should be equivalent
```

This above messes without the output, as now the test suite is titled I am a comment because of the tape like syntax tap-out looks for.

### CODE REVIEW

**Comment by dhershman1** [comment]
Not ready to push for this solution just yet. I will probably work on the feature for #24 extensions and add that in as a minor version update (granted it doesn't break anything ofc)
